app/routes/items/new.js

```diff
@@ -143,29 +148,38 @@ export default AuthorizeRoute.extend(GradeMixin, {
     controller.set("height", null);
     controller.set("weight", null);
     controller.set("pieces", null);
-    controller.set("expirty_date", null);
-    controller.set("selectedGrade", {
-      name: "B",
-      id: "B"
-    });
+    controller.set("expiry_date", null);
     controller.set("imageKeys", "");
 
-    controller.set(
-      "restrictionId",
-      this.get("restrictionOptions").get("firstObject")
-    );
-    controller.set(
-      "saleableId",
-      this.get("saleableOptions").get("firstObject")
-    );
-    const defaultValue = await this.get("packageService").getItemValuation({
-      donorConditionId: this.getDefaultCondition().id,
-      grade: controller.get("selectedGrade").id,
-      packageTypeId: controller.get("codeId")
-    });
-
-    controller.set("defaultCondition", this.getDefaultCondition());
-    controller.set("valueHkDollar", +defaultValue.value_hk_dollar);
-    controller.set("defaultValueHkDollar", +defaultValue.value_hk_dollar);
+    if (["Box", "Pallet"].indexOf(storageType) < 0) {
```

Comment:
initializeAttribute was a pretty big method. now with this new check, its lines has grown too much. Can you just split out into separate methods

something like

```javascript
 async initializeAttributes(controller, model) {
   const storageType = this.paramsFor("items.new").storageType;
   const isBoxPallet = ["Box", "Pallet"].indexOf(storageType);
   this.initializeSubFormAttributes(isBoxPallet, controller)
   this.initializeSomethingElse(isBoxPallet, controller)
}
```
and move all the if else logic to its respective method
It will maintain/improve the readability